Source/NSFontCollection.m

```diff
@@ -0,0 +1,341 @@
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSError.h>
+#import <Foundation/NSValue.h>
+#import <Foundation/NSDictionary.h>
+#import <AppKit/NSFontCollection.h>
+#import <GNUstepGUI/GSFontInfo.h>
+
+// NOTE: Some of this cannot be implemented currently since the backend does not support physically
+//       moving fonts on the filesystem...  this is here for compatilibility for now.
+
+@interface NSMutableFontCollection (Private)
+
+- (void) _setFonts: (NSArray *)fonts;
+- (void) _setQueryAttributes: (NSArray *)queryAttributes;
+
+@end
+
+@implementation NSMutableFontCollection (Private)
+
+- (void) _setFonts: (NSArray *)fonts
+{
+  [_fonts addObjectsFromArray: fonts];
+}
+
+- (void) _setQueryAttributes: (NSArray *)queryAttributes
+{
+  ASSIGN(_queryAttributes, [queryAttributes mutableCopy]);
+}
+
+@end
+
+@interface NSCTFontCollection : NSObject
+@end
+
+@implementation NSCTFontCollection
+@end
+
+@implementation NSFontCollection 
+
+static NSMutableDictionary *__sharedFontCollections;
+static NSMutableDictionary *__sharedFontCollectionsVisibility;
+static NSMutableSet *__sharedFontCollectionsHidden;
+
++ (void) initialize
+{
+  if (self == [NSFontCollection class])
+    {
+      __sharedFontCollections = [[NSMutableDictionary alloc] initWithCapacity: 100];
+      __sharedFontCollectionsVisibility = [[NSMutableDictionary alloc] initWithCapacity: 100];
+      __sharedFontCollectionsHidden = [[NSMutableSet alloc] initWithCapacity: 100];
+
+      [__sharedFontCollections setObject: [NSFontCollection fontCollectionWithAllAvailableDescriptors]
+                                  forKey: NSFontCollectionAllFonts];
+    }
+}
+
+// Initializers...
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _fonts = [[NSMutableArray alloc] initWithCapacity: 50];
+      _queryDescriptors = [[NSMutableArray alloc] initWithCapacity: 10];
+      _exclusionDescriptors = [[NSMutableArray alloc] initWithCapacity: 10];
+      _queryAttributes = [[NSMutableArray alloc] initWithCapacity: 10];
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_fonts);
+  RELEASE(_queryDescriptors);
+  RELEASE(_exclusionDescriptors);
+  RELEASE(_queryAttributes);
+  [super dealloc];
+}
+
+// This method will get the actual list of fonts
+- (void) _runQueryWithDescriptors: (NSArray *)queryDescriptors
```

Comment:
Looks like we do not need this method. But we need something similar that turns logical font descriptors into physical ones. (Maybe based on NSFontDescriptor matchingFontDescriptorsWithMandatoryKeys:?) This should also respect the _exclusionDescriptors of course.